test/Conversion/LTLToCore/assertproperty.mlir

```diff
@@ -0,0 +1,42 @@
+// RUN: circt-opt %s --lower-ltl-to-core | FileCheck %s
+// Tests that an Assert Property high level statement can be converted correctly
+
+module {
+  //CHECK:  hw.module @test(in %clock : !seq.clock, in %reset : i1, in %0 "" : i1)
+  hw.module @test(in %clock : !seq.clock, in %reset : i1, in %8 : i1) {
+    //CHECK:  %1 = seq.from_clock %clock
+    %0 = seq.from_clock %clock 
+    //CHECK:  %true = hw.constant true
+    %true = hw.constant true
+
+    //CHECK:  %false = hw.constant false
+    //CHECK:  %true_0 = hw.constant true
+    //CHECK:  %2 = comb.or %reset, %hbr : i1
+    //CHECK:  %hbr = seq.compreg sym @hbr  %2, %clock powerOn %false : i1  
+    %9 = verif.has_been_reset %0, sync %reset
+
+    //CHECK:  %3 = comb.xor %reset, %true_0 : i1
+    //CHECK:  %4 = comb.and %hbr, %3 : i1
+    //CHECK:  %5 = comb.xor bin %4, %true : i1
+    %10 = comb.xor bin %9, %true : i1
+
+    //CHECK:  %6 = hw.wire %0 : i1
+    %11 = hw.wire %8 : i1
+
+    //CHECK:  %7 = hw.wire %5 : i1
+    %12 = hw.wire %10 : i1
+
+    //CHECK:  %8 = comb.or %7, %6 : i1
+    %13 = ltl.disable %11 if %12 : i1
+
+    //CHECK:  sv.always posedge %1 {
+    //CHECK:    sv.assert %8, immediate
+    //CHECK:  }
+    %14 = ltl.clock %13, posedge %0 : !ltl.property
+    verif.assert %14 : !ltl.property
```

Comment:
Hmm this feels like the Verif dialect is missing an op to capture what you need. Do we need something like a `verif.clocked_assert` op, with an explicit clock operand? The grand vision for CIRCT is to have a set of core dialects that can capture everything you need without reaching for SV. The fact that you do have to use SV here suggests that we're missing something.

Does your BTOR2 work recognize `sv.always` and `sv.assert` to do its emission? If yes, would that become easier if this were a single `verif.clocked_assert` op?

Reply:
yes definitely, ideally I wouldn't have to look for use-def pairs to find the clock associated to the assertion. It would also be nice to have a semantic way to differentiate between property assertions (that need to be lowered) and regular assertions (that can be used as is) for the btor2 back-end. This is currently why I chose to do the lowering this way, if you and @uenoku both think that it would be better to leave the `verif.assert` and the `ltl.clock` here, then I can add their support in the btor2 back-end. Let me know what you think!

Reply:
Supporting `verif.assert` and `ltl.clock` in the btor2 back-end would definitely be great. I assume you have to do some analysis to detect `sv.always ... { sv.assert }` as clocked assert anyway, so loocking for `verif.assert(ltl.clock)` might be strictly easier/more idiomatic already.

The question is whether we want to make `verif.assert(ltl.clock)` the "official" way of having a clocked assertion in the core dialect. Or whether we want that explicit `clocked_assert` that you've PR'd. Let's see what others think :-)